packages/mendel-pipeline/src/cli.js

```diff
@@ -20,6 +20,30 @@ program
     .option('-w, --watch', 'Watch mode', false)
     .parse(process.argv);
 
-// Example usage
-const mendelConfig = config(Object.assign(program, {cwd: path.resolve(process.cwd(), program.args[0])}));
-mendelPipeline(mendelConfig);
+
+const cliOptions = Object.assign(program, {
+    cwd: path.resolve(process.cwd(), program.args[0]),
+});
+
+if (program.watch) {
+    const daemon = new MendelPipelineDaemon(cliOptions);
```

Comment:
See the `startServer` idea in the `daemon.js`, since there will be a default port, this would throw "ERRADDINUSE" and the program exists, no harm done.